### PR TITLE
Support a pipeline of middlewares to handle a message

### PR DESCRIPTION
**Pull request by danizord**
I managed to build a pipeline without Zend\Stratigility dependency. Yes the code is a bit hard to understand but I'm pretty confident it works (not tested yet).

The only drawback I can find of this approach is that it does not support Stratigility's error handling stuff (passing the 3rd $error argument to $next).

Should I drop it and add Stratigility dependency instead?

### CODE REVIEW

File: src/Middleware/WorkerMiddleware.php [hunk 1]
```diff
@@ -102,6 +102,28 @@ private function getMiddlewareForMessage(string $messageName): callable
...
+
+        if (!is_array($this->messagesMapping[$messageName])) {
+            // throw exception: must be array or string
+        }
+
+        // Instantiate middlewares
+        $middlewares = array_map([$this->container, 'get'], $this->messagesMapping[$messageName]);
+
+        // Build and return a simle pipeline
+        $pipeline = function ($req, $res, $out) use ($middlewares, &$pipeline) {
```
Comment (bakura10) [inline src/Middleware/WorkerMiddleware.php]: What's the intent of using a reference here on &pipline?

Comment (danizord) [inline src/Middleware/WorkerMiddleware.php]: Because this variable is not defined at this moment (we're declaring `$pipeline` in same line).

**Comment by bakura10** [comment]
Just thought about it, but a nice side-effect of this is that we won't need Event Manager anymore. If an app needs to do more things on an event, it just needs to register an additional middleware :D.

**Comment by danizord** [comment]
> Just thought about it, but a nice side-effect of this is that we won't need Event Manager anymore. If an app needs to do more things on an event, it just needs to register an additional middleware :D.

Kind of, we just need to make sure that some middleware will not return response without calling $next.

**Comment by danizord** [comment]
@bakura10 done!

PS: I'm deeply sorry to have not used Stratigility to dispatch the pipeline. Testing this stuff was the hardest thing ever :o

**Comment by bakura10** [comment]
Looks good. Thanks! :)

**Pull request merged**